packages/quantic/force-app/main/default/lwc/quanticSort/__tests__/quanticSort.test.js

```diff
@@ -145,14 +195,158 @@ describe('c-quantic-sort', () => {
 
   afterEach(() => {
     cleanup();
+    searchStatusState = defaultSearchStatusState;
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    afterAll(() => {
+      mockSuccessfulHeadlessInitialization();
+    });
+
+    it('should display the initialization error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
   });
 
   describe('controller initialization', () => {
-    it('should subscribe to the headless state changes', async () => {
+    it('should subscribe to the headless sort and search status state changes', async () => {
       createTestComponent();
       await flushPromises();
 
-      expect(functionsMocks.subscribe).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.sortStateSubscriber).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.searchStatusStateSubscriber).toHaveBeenCalledTimes(
+        1
+      );
+    });
+  });
+
+  describe('when no results are found', () => {
+    beforeAll(() => {
+      searchStatusState = {...defaultSearchStatusState, hasResults: false};
+    });
+
+    it('should not display the sort dropdown', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const sortDropdown = element.shadowRoot.querySelector(
+        selectors.sortDropdown
+      );
+
+      expect(sortDropdown).toBeNull();
+    });
+  });
+
+  describe('when a sort option is selected', () => {
+    it('should call the sortBy method of the sort controller', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const lightningCombobox = element.shadowRoot.querySelector(
+        selectors.lightningCombobox
+      );
+      const exampleDefaultSortOptionValue = 'relevance';
```

Comment:
If you want to be even more robust, you could go read the value of the the first sort option for example,
click on that first sort option, and then expect that value to be passed in the `change` event.

Reply:
Since this code is using the out of the box Lightning Combobox from Salesforce, I did not see the need to go that robust with the testing.

Reply:
True for unit tests indeed.